table/block_based/block.cc

```diff
@@ -394,6 +394,12 @@ void IndexBlockIter::Seek(const Slice& target) {
   bool ok = false;
   if (prefix_index_) {
     ok = PrefixSeek(target, &index);
+    if (!ok && !Valid()) {
+      // This is to let the caller to distinguish between non-existing prefix,
+      // and when key is larger than the last key, which both set Valid() to
+      // false.
+      status_ = Status::NotFound();
```

Comment:
I looked at the code a little bit and think we might need to return here. Here even if the key is larger than largest block key in the bucket, we still point to the last block. But if we execute line 414 to 421, I'm not sure where it will end up with. Since hash index doesn't work with restart offset more than 1, it should be fine.

Reply:
Line 412 does the return.

Reply:
My suggestion is to returning for ok case too.